src/main.rs

```diff
@@ -166,6 +168,18 @@ async fn run_single_threaded(relink: bool) -> anyhow::Result<()> {
     let mut last_render_at = Instant::now();
     let is_render_spawned = Arc::new(AtomicBool::new(false));
 
+    let tick_tx = tx.clone();
+    // Tick to trigger receipt sending
+    tokio::spawn(async move {
```

Comment:
There is `tokio::time::interval` which gives you a stream of ticks. This one then needs to be joined with `tx`. This is a little bit advanced. Feel free to leave it out!